docs/vf-setup.md

```diff
@@ -0,0 +1,141 @@
+# Setting up Virtual Functions
+
+The SR-IOV Network Device Plugin requires SR-IOV virtual functions (VFs) to be created on the host ahead of startup. The device plugin does not manage the creation of virtual functions, and it does not automatically update when there is a change in the virtual functions. Each time the number of VFs or the driver used by the VF change the SR-IOV Network device plugin needs to be restarted. 
+This guide gives an overview for setting up SR-IOV virtual functions on a linux host for various NICs. 
+
+  * [Intel](#intel)
+  * [Mellanox](#mellanox)
+
+## Intel
+The below works on intel SR-IOV enabled cards including those from the Gigabit Ethernet, X710 and X722 families of Network Interface Cards. Note not all Gigabit Ethernet NICs are enabled with SR-IOV.
+
+Cards from the following families can be enabled through the sysfs endpoint as described below:
+* **Intel i40e Driver**
+	* Intel® Ethernet Controller X722 
+	* Intel® Ethernet Controller X710
+* **Intel ixgbe Driver** 
+	* Intel® Ethernet Controller 82599
+	* Intel® Ethernet Controller X520
+	* Intel® Ethernet Controller X540
+	* Intel® Ethernet Controller x550
+	* Intel® Ethernet Controller X552
+	* Intel® Ethernet Controller X553
+
+### Creating VFs with sysfs
+First select a compatible NIC on which to create VFs and record its name (shown as PF_NAME below). 
+
+To create 8 virtual functions run:
+```
+echo 8 > /sys/class/net/${PF_NAME}/device/sriov_numvfs
```

Comment:
Should we reset the number of VFs here before asking for 8 in case it was configured previously and will throw an error if its not reset first before asking for a new set of VFs?

Reply:
Added a section under "common problems" to pick up this error if someone wants to reset the number of VFs